include/dlaf/lapack/tile.h

```diff
@@ -423,6 +454,7 @@ class CusolverInfo {
 
 public:
   CusolverInfo(int workspace_size) : workspace_(workspace_size), info_(1) {}
+  CusolverInfo() : info_(1) {}
```

Comment:
Question related to this and `rocsolver` not requiring a workspace:

Why does `MemoryView` allocate a `MemoryChunk` for the empty state (size 0, default-constructed, and moved-from) instead of just emptying the shared pointer? Is the behaviour relied on anywhere? Looking at the `operator()` overloads for `MemoryView` it seems like it would not need to `make_shared` a `MemoryChunk` since the size is anyways checked explicitly (https://github.com/eth-cscs/DLA-Future/blob/4bb4283fbf0eb344e5302d8ce1e626b84be990b4/include/dlaf/memory/memory_view.h#L130-L143).

The relation is that if we can change `MemoryView` to not allocate in the default constructor then we don't need to add `ifdef`s here to avoid the `MemoryChunk` allocation here in `CusolverInfo`.

Reply:
I've made some tentative changes to not allocate size-0 memory views. I can split that off into a separate PR as well if it's easier to review that way.

Reply:
Any comments on the general idea here, or the specific changes that I've implemented?

Reply:
I've `ifdef`ed out the workspace when HIP support is enabled, but I've left the `memory_view` changes for now. I'll move that out to a separate PR if requested.

Reply:
Yes, better to move it in a different PR.